keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -124,19 +73,122 @@ class ClusterHandler(
           "Upsert cluster ${spec.moniker.cluster} in ${spec.location.accountName}/${spec.location.region}",
           spec.moniker.application,
           "Upsert cluster ${spec.moniker.cluster} in ${spec.location.accountName}/${spec.location.region}",
-          listOf(Job("createServerGroup", job)),
-          OrchestrationTrigger(resourceName.toString())
+          listOf(Job(job["type"].toString(), job)),
+          OrchestrationTrigger(resource.metadata.name.toString())
         ))
         .await()
     }
     log.info("Started task {} to upsert cluster", taskRef.ref)
   }
 
-  override fun delete(resourceName: ResourceName, spec: SecurityGroup) {
+  /**
+   * @return `true` if the only changes in the diff are to capacity.
+   */
+  private fun Diff?.isCapacityOnly(): Boolean =
+    this != null && changes.all {
+      it.affectedObject.run {
+        isPresent && get() is Capacity
+      }
+    }
```

Comment:
I like that this check is strongly typed (because the objects being diffed were).